compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/ComposeUiTestTest.kt

```diff
@@ -158,6 +166,31 @@ class ComposeUiTestTest {
         }
     }
 
+    @Test
+    fun testTimeout() {
+        val timeout = 100.milliseconds
+        val error = assertFailsWith<AssertionError> {
+            runComposeUiTest(testTimeout = timeout) {
+                // switch a dispatcher to not skip the delay
+                withContext(Dispatchers.Default) { delay(1000) }
+            }
+        }
+        assertTrue(
+            error.message!!.startsWith("After waiting for $timeout, the test coroutine is not completing")
```

Comment:
I don't know if testing specific error messages is a good idea. There's no reason for our test to fail if the library changes the message in the future.

Reply:
There are no asserts in the code, so it seems that if it fails, it can only be due to timeout, no?

Reply:
> There are no asserts in the code, so it seems that if it fails, it can only be due to timeout, no?

The test doesn't "know" the internals of the runTest implementation. There could be a simple `assert(...)` call unrelated to the test block under the hood which will throw an `AssertionError `. 

I'd prefer a test to fail due to a change of the mesage (can be fixed by updating the expected message along with the library version update), than if a test would accidentaly pass due to a change in our implementations (a new `assert` added for some reason).

Reply:
/**
 * Thrown when a test has completed and there are tasks that are not completed or cancelled.
 */
@ExperimentalCoroutinesApi
internal class UncompletedCoroutinesError(message: String) : AssertionError(message)

The exact assertionerror type is internal. We can assert that the class qualified name is UncompletedCoroutinesError. Wdyt?

Reply:
I'm not sure it's legitimate for an assert to be present like that in non-user code. If it existed, it would mean an empty test could fail.

Reply:
I don't know if that's better, but it's not so important that we should continue working on this. Let's just leave it at the current state.